backend/dataall/modules/shares_base/tasks/share_verifier_task.py

```diff
@@ -37,8 +39,18 @@ def verify_shares(engine):
         return processed_share_objects
 
 
+def trigger_reapply_task():
```

Comment:
By only triggering the reapply task if unhealthy shares are detected, we can avoid unnecessary ECS task executions.
We could have something like: 
```
def trigger_reapply_task(should_trigger):
    if should_trigger:
        log.info("Unhealthy shares detected, triggering reapply task...")
        Ecs.run_ecs_task(
            task_definition_param='ecs/task_def_arn/share_reapplier',
            container_name_param='ecs/container/share_reapplier',
            context=[],
        )
    else:
        log.info("No unhealthy shares detected, skipping reapply task.")
```

And define should_trigger in verify_shares by fetching health status of items.

Reply:
we currently don't return that information from `verify_shares` and don't have an straightforward mechanism to determine if there are unhealthy shares other than adding a new query to find all ShareObjectItems that have healthStatus of `Unhealthy`

While we can do that and populate the value of `should_trigger` to T/F based on the count of that query - I personally feel that the additional cost is very negligible and there is likely no need

ECS pricing model is variable per GB and CPU down to a per second granularity - and our ECS Task uses 1vCPU and 2GB of memory

Especially, since these tasks a scheduled on a weekly cadence - the additional pricing if any would be fractions of cents

Link to ECS Fargate Pricing model - https://aws.amazon.com/fargate/pricing/

If others feel strongly I can re-evaluate!